app/controllers/RiderStageConnectionController.java

```diff
@@ -66,19 +65,15 @@ public RiderStageConnectionController(StageRepository stageRepository, RiderRepo
     }
 
     @BodyParser.Of(BodyParser.Json.class)
-    public CompletableFuture<Result> addStage() {
+    public CompletionStage<Result> updateRiderStageConnection(long riderStageConnectionId) {
         JsonNode json = request().body().asJson();
-        return parseStage(json).thenApply(stage -> stageRepository.addStage(stage)).thenApply(message -> {
-            return ok(message + " has been added");
+        return parseRiderStageConnection(json, riderStageConnectionId).thenApply(riderStageConnectionRepository::updateRiderStageConnection).thenApplyAsync(rSC -> {
+            return ok("success");
         }).exceptionally(ex -> {
             Result res = null;
-            String name = ExceptionUtils.getRootCause(ex).getClass().getSimpleName();
             switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)